llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -971,6 +985,10 @@ class VPRecipeWithIRFlags : public VPRecipeBase {
       I->setHasAllowContract(FMFs.AllowContract);
       I->setHasApproxFunc(FMFs.ApproxFunc);
       break;
+    case OperationType::CastOp:
+      if (isa<ZExtInst>(I))
+        I->setNonNeg(CastFlags.NonNeg);
```

Comment:
For other flags, we only set OpType if the underlying operant supports the flag, so no extra checks are needed. To be consistent with other flags, it would be good to avoid the need for checking `I` here, by having `ZExtFlags` /`ZExtOp` instead here.

Reply:
See PossiblyNonNegInst

Reply:
Thanks, NonNegFlags/NonNegOp would probably be best